### DISCUSSION TITLE
Is there a way I can install a third party nginx module for dokku to use?

### DISCUSSION THREAD

**Discussion by insanity54** [dokku/dokku#6004] [Q&A] [answered]
I am using nginx as dokku's proxy server. 

I have an app in which I need to protect an API endpoint with JWT authentication. Is there a way to do this using dokku?

I found https://github.com/TeslaGov/ngx-http-auth-jwt-module. Is there a way I can build a custom nginx with ngx-http-auth-jwt-module, and have dokku use that custom nginx instead of the regular nginx? 

Is this a bad idea? Is there a better way to achieve what I'm looking for?

Edit: I found a similar issue in the tracker. https://github.com/dokku/dokku/issues/2780

**Comment by josegonzalez** [comment, marked as answer]
Dokku just uses system nginx, so if you compile an nginx binary on the system with your custom modules, it should work just fine.

Compiling a custom system package is outside of the scope of dokku, so I would suggest doing research on how to do that (digitalocean may have tutorials on this).
